### DISCUSSION TITLE
Detecting if a window (or frame) is full screen or not?

### DISCUSSION THREAD

**Discussion by barries** [emilk/egui#5111] [Q&A] [answered]
I'm starting with the `eframe_template` app for a native app for Linux and Windows, and trying to make F11 toggle fullscreen / not full screen. How can I detect whether it is in fullscreen mode, so I can toggle it?

I've tried `frame.info().window_info.fullscreen` based on an old answer, but `IntegrationInfo` doesn't have a `window_info` field, and looking through the docs, I wasn't able to find how to get it.

Thanks!

**Comment by KatchaVlad** [comment]
I access the viewport state to get that information, I don't know if there is a cleaner way :
`ui.input(|i| i.viewport().fullscreen)`

**Comment by rustbasic** [comment, marked as answer]
The method mentioned above is correct.

```
let fullscreen = ui.input(|i| i.viewport().fullscreen.unwrap_or(false));
let fullscreen = ui.ctx().input(|i| i.viewport().fullscreen.unwrap_or(false));
let fullscreen = ctx.input(|i| i.viewport().fullscreen.unwrap_or(false));
```

**Comment by barries** [comment]
Thanks! Here's what I ended up with

```
if let Some(new_fullscreen) = ctx.input(|i| {
    if i.key_pressed(Key::F11) {
        Some(!i.viewport().fullscreen.unwrap_or(false))
    } else {
        None
    }
}) {
    ctx.send_viewport_cmd(egui::ViewportCommand::Fullscreen(new_fullscreen));
}
```

The only remaining bug is that `egui` still renders its window decorations in full screen mode.

![image](https://github.com/user-attachments/assets/f25c3e1e-1667-4fcf-8fdd-eb45bf7339f4)

**Reply by barries** [reply]
Ok, removing the decorations "by handle" fixed it.

```
if let Some(new_fullscreen) = ctx.input(|i| {
    if i.key_pressed(Key::F11) { Some(!i.viewport().fullscreen.unwrap_or(false)) }
    else                       { None                                            }
}) {
    ctx.send_viewport_cmd(egui::ViewportCommand::Fullscreen(  new_fullscreen));
    ctx.send_viewport_cmd(egui::ViewportCommand::Decorations(!new_fullscreen));
}
```
